src/templates.rs

```diff
@@ -0,0 +1,224 @@
+use std::collections::HashMap;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use crate::{Credentials, MailgunResult};
+
+const TEMPLATES_ENDPOINT: &str = "templates";
+const TEMPLATE_VERSIONS_ENDPOINT: &str = "versions";
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateTemplateResponse {
+    pub message: String,
+    pub template: TemplateResponse,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GetTemplatesResponse {
+    pub items: Vec<TemplateResponse>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GetSingleTemplateResponse {
+    pub template: TemplateResponse,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct TemplateResponse {
+    pub created_at: String,
+    pub created_by: String,
+    pub description: String,
+    pub name: String,
+    pub id: String,
+    pub version: Option<VersionResponse>,
+    pub versions: Option<Vec<VersionResponse>>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct VersionResponse {
+    created_at: String,
+    engine: String,
+    tag: String,
+    comment: String,
+    mjml: String,
+    template: Option<String>,
+    id: Option<String>,
+    active: bool,
+}
+
+#[derive(Serialize, Default, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Template {
+    pub name: String,
+    pub description: String,
+    pub template: Option<String>,
+    pub tag: Option<String>,
+    pub engine: Option<String>,
```

Comment:
could we make this a single-value enum so that it's only possible to pass in `handlebars` for now?